### PR TITLE
Data Science with Python : Random Forest Algorithm

### LINKED ISSUE #1038: Data Science with Python : Random Forest Algorithm
Welcome to 'DSWP' Team, good to see you here.

This issue will helps readers in acquiring all the knowledge to learn about getting involved with Random Forest Algorithm. Tutorial to Random Forest Algorithm and how it's applied using sample code.

To get assigned to this issue, add your serial numbers mentioned in the spreadsheet of "Data Science with Python", the approach one would follow and choice you prefer (Documentation, Audio, Video). You can go with all three or any number of options you're interested to work on.

This issue is created for beginners and who aren't assigned to any issues yet.

If you had referred any resources, add them up in "DS Resources". Similarly if you had used datasets, include them in "DS Datasets".

Domain : Machine Learning

Mentors Assigned : Aishani Singh / Gauri Bharadwaj

Points to Note :

The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.

"Issue Title" and "PR Title should be the same. Include issue number along with it.

Changes should be made inside the Datascience_With_Python/ directory & Datascience_With_Python branch.

Follow Contributing Guidelines & Code of Conduct before start Contributing.

This issue is only for 'GWOC' contributors of 'DSWP' domain.

All the best. Enjoy your open source journey ahead. 😎

### PR DESCRIPTION
**Pull request by Akshat2019VITB**
<hr>

## Description 📜

Provided the documentation for Random Forest Algorithm with explanation on data cleaning and training the ML model.
Fixes #1038

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] Datascience with Python

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

### CODE REVIEW

**Comment by manthan89-py** [comment]
Interested for this work

**Comment by prathimacode-hub** [comment]
Add up your issue number? @Akshat2019VITB

**Comment by Akshat2019VITB** [comment]
Issue Number - 1038

**Review by prathimacode-hub — CHANGES_REQUESTED** [review]
Changes to be done :
 
The readme file should be titled as README.md and project file name in snake case : random_forest_algorithm.ipynb
In README.md, add workflow, its usage, advantages, applications, why its preferred, final outcomes and conclusion.
Instead of getting directly into code implementation. start with introduction of the issue topic and then into coding and present overall summary of your work done at the end.

Kindly work on it. @Akshat2019VITB

File: Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+- E-Commerce
+
+## **Workflow**
+
+- First we start by collection of our data, for this we have chosen a dataset for Social Network Ads.
+
+- Then we continue on by reading the data, as shown.
+
+![](https://github.com/Akshat2019VITB/winter-of-contributing/blob/Datascience_With_Python/Datascience_With_Python/Machine%20Learning/Algorithms/Random%20Forest%20Algorithm/Images/5.PNG)
```
Comment (DhruvBajaj01) [inline Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md:69]: You need not add a screenshot is read me. Just an overview will work.

File: Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+![](https://github.com/Akshat2019VITB/winter-of-contributing/blob/Datascience_With_Python/Datascience_With_Python/Machine%20Learning/Algorithms/Random%20Forest%20Algorithm/Images/7.PNG)
+
+- After the processsing of data, we can now split the data into training and testing data and apply Random Forest Classifier to train our model.
+
+- Evaluation of our model will be done with an accuracy score as shown in the screenshots.
+
+## **Understanding a Boxplot**
+
+![](https://github.com/Akshat2019VITB/winter-of-contributing/blob/Datascience_With_Python/Datascience_With_Python/Machine%20Learning/Algorithms/Random%20Forest%20Algorithm/Images/9.png)
+
```
Comment (DhruvBajaj01) [inline Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md:90]: You need not explain your code here its, already in your notebook.

File: Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+- We saw it's advantages and how it is useful with its features and it's advantage over Decision Tree.
+
+- Data preprocessing was done to check for missing values and outliers, which were removed through Z-Score threshold.
+
+- Label Encoding was used to convert categorical data into numerical.
+
+- Data was split into Training and Testing for our ML model.
+
+- We were able to get an accuracy score of 0.95 from our model through testing.
+
```
Comment (DhruvBajaj01) [inline Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md:122]: Add libraries that you used while constructing the project.

**Review by DhruvBajaj01 — COMMENTED** [review]
I will go through the code, once you make you changes in your readme. Thank you for your contribution.

**Comment by Akshat2019VITB** [comment]
@DhruvBajaj01 I have made the changes in the README.md file as requested.

**Comment by DhruvBajaj01** [comment]
Have you removed code images from the readme file? Also, have you added libraries used?

**Comment by Akshat2019VITB** [comment]
I have added the libraries used and removed the code images. @DhruvBajaj01

File: Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
+# **Random Forest Algorithm**
+
+## **Introduction**
+
+- A random forest is a machine learning technique that’s used to solve regression and classification problems. It utilizes ensemble learning, which is a technique that combines many classifiers to provide solutions to complex problems.
```
Comment (DhruvBajaj01) [inline Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md:5]: Plagiarised Line

File: Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+- We will understand the working of this algorithm by a project example by following steps as below:
+
+   1. Data Preprocessing with Visualization.
+   2. Model Building
+
+## **Purpose** 
+
+The (random forest) algorithm establishes the outcome based on the predictions of the decision trees. It predicts by taking the average or mean of the output from various trees. Increasing the number of trees increases the precision of the outcome.
```
Comment (DhruvBajaj01) [inline Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md:14]: plagiarised line

File: Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+## **Brief summary** 
+
+A decision tree consists of three components: decision nodes, leaf nodes, and a root node. A decision tree algorithm divides a training dataset into branches, which further segregate into other branches. This sequence continues until a leaf node is attained. The leaf node cannot be segregated further.
```
Comment (DhruvBajaj01) [inline Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md:18]: plag

File: Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+A random forest consists of many such decision trees. 
+
+## **Features**
+
+- It’s more accurate than the decision tree algorithm.
```
Comment (DhruvBajaj01) [inline Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md:23]: plag

File: Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+- It provides an effective way of handling missing data.
+
+- It can produce a reasonable prediction without hyper-parameter tuning.
```
Comment (DhruvBajaj01) [inline Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md:27]: plag

File: Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+
+- It solves the issue of overfitting in decision trees.
```
Comment (DhruvBajaj01) [inline Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md:29]: plag

File: Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+
+- In every random forest tree, a subset of features is selected randomly at the node’s splitting point.
+
+## **Advantages**
+
+- Random forests is considered as a highly accurate and robust method because of the number of decision trees participating in the process.
```
Comment (DhruvBajaj01) [inline Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md:35]: plag

File: Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+
+- It does not suffer from the overfitting problem. The main reason is that it takes the average of all the predictions, which cancels out the biases.
```
Comment (DhruvBajaj01) [inline Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md:37]: plag

File: Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+
+- The algorithm can be used in both classification and regression problems.
```
Comment (DhruvBajaj01) [inline Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md:39]: plag

File: Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+
+- Random forests can also handle missing values. There are two ways to handle these: using median values to replace continuous variables, and computing the proximity-weighted average of missing values.
```
Comment (DhruvBajaj01) [inline Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md:41]: plag

File: Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+
+- You can get the relative feature importance, which helps in selecting the most contributing features for the classifier.
```
Comment (DhruvBajaj01) [inline Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md:43]: plag

File: Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+
+## **Random Forest vs Decision Tree**
+
+- Random forests is a set of multiple decision trees.
```
Comment (DhruvBajaj01) [inline Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md:47]: plag

File: Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+
+- Deep decision trees may suffer from overfitting, but random forests prevents overfitting by creating trees on random subsets.
```
Comment (DhruvBajaj01) [inline Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md:49]: plag

File: Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md [hunk 17]
```diff
@@ -0,0 +1,120 @@
...
+
+- Label Encoding was used to convert categorical data into numerical.
+
+- Data was split into Training and Testing for our ML model.
+
+- We were able to get an accuracy score of 0.95 from our model through testing.
+
+## **Conclusion**
+
+- The Random Forest Algorithm is a machine learning algorithm that is easy to use and flexible. It uses ensemble learning, which enables organizations to solve regression and classification problems.
```
Comment (DhruvBajaj01) [inline Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md:118]: plag

File: Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md [hunk 18]
```diff
@@ -0,0 +1,120 @@
...
+
+- This is an ideal algorithm for developers because it solves the problem of overfitting of datasets. It’s a very resourceful tool for making accurate predictions needed in strategic decision making in organizations.
```
Comment (DhruvBajaj01) [inline Datascience_With_Python/Machine Learning/Algorithms/Random Forest Algorithm/README.md:120]: plag

**Review by DhruvBajaj01 — COMMENTED** [review]
I added plag in front of the line that was plagiarised. Your plagiarism counts to around 51% try to reduce it to at least 20 %. 
Go through https://www.duplichecker.com/ before submitting your file.
Make the changes and you are ready to merge PR

**Comment by Akshat2019VITB** [comment]
@DhruvBajaj01 I have the required changes in README.md.

**Review by DhruvBajaj01 — COMMENTED** [review]
Nice work @Akshat2019VITB, @prathimacode-hub please make a final review and merge the contribution.

**Review by prathimacode-hub — APPROVED** [review]
@Akshat2019VITB Well done. You had done a amazing work. 🎉👌

**Pull request merged**